### ISSUE TITLE
Connection timeouts when using Spigot 1.21.3

### ISSUE THREAD

**Issue by jacob1** [DiscordSRV/DiscordSRV#1776]
### Expected behavior

Using the latest snapshot of this plugin with 1.21.3 support, it shouldn't interfere with server connections.

### Actual behavior

Attempting to join the server, the connection will hang for about 10-20 seconds, and then give up with the message "ReadTimeoutException : null" (or another connection timeout error). This prevents joining the server at all.

This will ONLY happen after `Server empty for 60 seconds, pausing` has been printed to the console log. DiscordSRV doesn't like something relating to the server pausing.

### Steps to reproduce

Setup a fresh Spigot 1.21.3 server, with DiscordSRV as the only plugin. Start it up, and wait 60 seconds. Don't join during this time, the connection hang only happens if there's been no successful joins before. After the server pauses, you're now unable to join.

I ran `/discordsrv debugger` in the console (as I couldn't join) and got [this output](https://bin.scarsz.me/0e7e67e7-58ce-465c-aefa-9f697b68e791#l3LZKnlFe8XiI0k3JO9o0aCkfa6QGplE)
Here is [latest.log](https://bin.scarsz.me/4c799de0-e20e-4d8c-9b43-306e54cc73cc#sfVNsYffRAaTrP3yGYMXkUgqHDX4qq0Q)

I'd guess there's some kind of thread hung up, but I don't see anything obvious in the logs so I'm at a loss to what causes this. Reproducing it is 100% consistent though.

This issue DOES NOT happen when using Paper, only Spigot. I'd guess that's because Paper worked around it here: https://github.com/PaperMC/Paper/pull/11526 . 

### Server software and version

CraftBukkit version 4376-Spigot-e65d67a-2702c5c (MC: 1.21.3) (Implementing API version 1.21.3-R0.1-SNAPSHOT)

### Checks

- [X] I am __not__ using an outdated version of DiscordSRV.
- [X] I asked in [DiscordSRV's Discord server](https://discord.discordsrv.com) to see whether this issue is in fact a bug that needs to be fixed.

### Ticket number or message link to where you asked in DiscordSRV's Discord server about this

19658

**Comment by Vankka** [comment]
Going by the Paper PR description the issue is probably the player pre login event, which DiscordSRV uses.

Would need to confirm but looks like a Spigot bug

**Comment by Vankka** [comment]
Also this relates to the `pause-when-empty-seconds` option in the `server.properties` file

**Comment by jacob1** [comment]
Yeah, on my server I disabled that option to prevent issues with this and any other plugins
